src/app/Mappings/components/MappingBuilder/MappingBuilder.tsx

```diff
@@ -30,7 +30,16 @@ export const MappingBuilder: React.FunctionComponent<IMappingBuilderProps> = ({
   builderItems,
   setBuilderItems,
 }: IMappingBuilderProps) => {
-  const reset = () => setBuilderItems([{ source: null, target: null }]);
+  const [isAddSourceDisabled, toggleAddMappingDisable] = React.useReducer(
+    (isToggled) => !isToggled,
+    false
+  );
```

Comment:
We don't need to have a separate state value for whether the button is disabled, since that can be derived from observing the `builderItems`. Keeping it separate and trying to sync the two could cause some bugs.